### PR TITLE
[Recipe] Spatial LibriSpeech

### PR DESCRIPTION
**Pull request by JinZr**
Support: [Spatial LibriSpeech](https://doi.org/10.21437/Interspeech.2023-2117)

Spatial LibriSpeech, is a spatial audio dataset with over 650 hours of first-order ambisonics, and optional distractor noise (with raw 19-channel audio coming soon). Spatial LibriSpeech is designed for machine learning model training, and it includes labels for source position, speaking direction, room acoustics and geometry. Spatial LibriSpeech was generated by augmenting LibriSpeech samples with 200k+ simulated acoustic conditions across 8k+ synthetic rooms.

### CODE REVIEW

File: lhotse/recipes/spatial_librispeech.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+from typing import Dict, Optional, Sequence, Union
+
+from tqdm.auto import tqdm
+
+from lhotse import fix_manifests, validate_recordings_and_supervisions
+from lhotse.audio import RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, resumable_download
+
+try:
```
Comment (pzelasko) [inline lhotse/recipes/spatial_librispeech.py:12]: Please move those try/except to local imports instead (inside download/prepare functions). Otherwise anybody importing lhotse without these dependencies will get these exceptions.

File: setup.py [hunk 2]
```diff
@@ -203,6 +203,10 @@ def mark_lhotse_version(version: str) -> None:
     "black==22.3.0",
     "isort==5.10.1",
     "pre-commit>=2.17.0,<=2.19.0",
+    "pandas>2.0.0,<2.3.0",
+    "pyarrow==17.0.0",
+    "fastparquet>2023.10.0,<2024.6.0",
+    "requests==2.31.0",
```
Comment (pzelasko) [inline setup.py:209]: Revert these changes, all of those should be optional as they are only needed for this recipe.

**Review by pzelasko — CHANGES_REQUESTED** [review]
Thanks for the contribution! I left a couple of comments.

**Comment by JinZr** [comment]
thank you for reviewing!

the concerns are solved now

Best Regards
Jin

**Review by pzelasko — APPROVED** [review]
Thanks! LGTM
